### ISSUE TITLE
tab to complete?

### ISSUE THREAD

**Issue by Nigelli** [LeaVerou/awesomplete#16832]
Hi, not necessarily an issue but as a user I expect to be able to select an item from the list by hitting the tab key.

Easy to fix by adding key code 9 to line 72 in awesomeplete.js?

`if ((c === 13 || c === 9) && me.selected) { // Enter or Tab`

**Comment by TxHawks** [comment]
That's not how a `select` element works...

`tab` moves you away from the widget, the arrow keys are used for selection

**Comment by Nigelli** [comment]
Yes of course, but once you have you choice selected if you tab to move away should this choice not be chosen?

Isn't it logical to assume that if the user has made the effort to go down the list and highlight the most appropriate item then it is unlikely their intention when tabbing away will be to cancel their choice?

**Comment by LeaVerou** [comment]
> Yes of course, but once you have you choice selected if you tab to move away should this choice not be chosen?
> 
> Isn't it logical to assume that if the user has made the effort to go down the list and highlight the most appropriate item then it is unlikely their intention when tabbing away will be to cancel their choice?

That's a very good point. I'm convinced. We should not `preventDefault()` on tab, but it should still make a selection.

**Comment by LeaVerou** [comment]
Although it would be a problem if the dropdown opens automatically, so maybe not then.

**Comment by Nigelli** [comment]
Sorry I might be missing something obvious but i'm not sure what you mean? If autoFirst is set to false and the user tabs nothing will be selected and therefore, tab will just move the user away.

If autoFirst is true then yes tabbing away would select the first option in the list. But isn't that the point of autoFirst? Using escape would still cancel this.

**Comment by LeaVerou** [comment]
Yes, I meant when using `autoFirst` as well.

**Comment by Frazer** [comment]
using tab to select and then stay in the field is not quite right either,  may I suggest:

    if (c === 13 && me.selected) { // Enter
        evt.preventDefault();
        me.select();
    }else if ( c === 9 && me.selected) { //Tab - dont prevent default
        me.select();
    }else....

**Comment by kball** [comment]
Just ran into this as well - looks like there was some movement in #17032 but it got tangled up with a lot of other changes and didn't happen. @LeaVerou would you accept a targeted PR that implements just this behavior behind an option flag?

**Comment by LeaVerou** [comment]
@kball If the code is reasonably good & compact, sure!

**Comment by kball** [comment]
@LeaVerou: Submitted in #17145. Came out pretty clean & compact, but let me know if you think anything needs fixing and I'll happily update

**Comment by paulopaquielli** [comment]
+1 Awaiting for release :)
